### PR TITLE
add: if for scipy and linting

### PR DESCRIPTION
**Pull request by 3sky**
The most important part is:
```bash
python3 -c '
import sys
version = sys.version_info
if version.major == 3 and version.minor == 11:
    print("pip3 install --no-deps --upgrade scipy==1.10.0")
else:
    print("pip3 install --no-deps --upgrade scipy==1.8.0")
' | sh
```
It's needed during working with python 3.11. 
Tests:
```bash
kuba@mace:~$ docker run -it --rm --entrypoint bash python:3.10
root@e56bfe8135cb:/# python3 -c '
import sys
version = sys.version_info
if version.major == 3 and version.minor == 11:
    print("pip3 install --no-deps --upgrade scipy==1.10.0")
else:
    print("pip3 install --no-deps --upgrade scipy==1.8.0")
' | sh
Collecting scipy==1.8.0
  Downloading scipy-1.8.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (39.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 39.0/39.0 MB 40.4 MB/s eta 0:00:00
Installing collected packages: scipy
Successfully installed scipy-1.8.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 23.0.1 -> 24.0
[notice] To update, run: pip install --upgrade pip
root@e56bfe8135cb:/#
exit
kuba@mace:~$ docker run -it --rm --entrypoint bash python:3.11
root@09f14ef087df:/# python3 -c '
import sys
version = sys.version_info
if version.major == 3 and version.minor == 11:
    print("pip3 install --no-deps --upgrade scipy==1.10.0")
else:
    print("pip3 install --no-deps --upgrade scipy==1.8.0")
' | sh
Collecting scipy==1.10.0
  Downloading scipy-1.10.0-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (110 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 110.1/110.1 kB 3.3 MB/s eta 0:00:00
Downloading scipy-1.10.0-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (30.7 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 30.7/30.7 MB 46.7 MB/s eta 0:00:00
Installing collected packages: scipy
Successfully installed scipy-1.10.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@09f14ef087df:/#
```

### CODE REVIEW

File: setup_deb.sh [hunk 1]
```diff
@@ -68,133 +68,142 @@ log "Installing python dependencies ..."
...
+    test-tube \
+    streamlit \
+    einops==0.3.0 \
+    webdataset==0.2.5 \
+    open-clip-torch==2.7.0 \
+    gradio==3.13.2 \
+    kornia==0.6 \
+    invisible-watermark \
+    streamlit-drawable-canvas==0.8.0 \
+    safetensors >=0.1.1 >=0.7.5 >=0.73.1 >=0.1.5 >=0.3.1
```
Comment (dkupnicki) [inline setup_deb.sh:184]: Looks like version specifiers of a few different packages got moved into this line

File: setup_deb.sh [hunk 2]
```diff
@@ -68,133 +68,142 @@ log "Installing python dependencies ..."
...
+    cchardet==2.1.7
+
+python3 -c '
+import sys
+version = sys.version_info
+if version.major == 3 and version.minor == 11:
+    print("pip3 install --no-deps --upgrade scipy==1.10.0")
+else:
+    print("pip3 install --no-deps --upgrade scipy==1.8.0")
+' | sh
```
Comment (dkupnicki) [inline setup_deb.sh:102]: I'm not a fan of this. I think we should instead move all Python requirements to a requirements.txt file and use environment markers in cases like this https://peps.python.org/pep-0508/#environment-markers

File: setup_deb.sh (hunk 1 above)
Comment (3sky) [reply setup_deb.sh:184]: Good catch, missed that.

File: setup_deb.sh (hunk 2 above)
Comment (3sky) [reply setup_deb.sh:102]: Yes, I agree. Using requirements.txt with 'python_version' marker will be much more elegant solution. I'm using it in manifests repo. However, it's up to you, If I can make this change, as it's big one.

Comment (dkupnicki) [reply setup_deb.sh:102]: I think the behaviour should stay the same if we keep all the flags and versions the same, so it wouldn't be that big of a change. @jan-grzybek-ampere what do you think?

**Review by dkupnicki — APPROVED** [review]

File: setup_deb.sh [hunk 3]
```diff
@@ -49,152 +49,48 @@ apt-get update -y
...
-   webdataset==0.2.5 \
-   open-clip-torch==2.7.0 \
-   gradio==3.13.2 \
-   kornia==0.6 \
-   invisible-watermark>=0.1.5 \
-   streamlit-drawable-canvas==0.8.0 \
-   safetensors>=0.3.1
+
+# get almost all python deps
+pip3 install --no-deps --upgrade -r requirments.txt
```
Comment (dkupnicki) [inline setup_deb.sh:71]: Looks like a typo here, but other than that I think everything looks fine

**Pull request merged**